File: JUNGOL/Beginner/#3699 변장/변장.cpp
#include <iostream>
#include <vector>
#include <map>
#include <string>
using namespace std;

int main() {
	int tc; cin>>tc;
	for(int i=0; i<tc; i++){
		int n; cin>>n;

		map<string, int> cnt;
		for(int j=0; j<n; j++){
			string t,s; cin>>t>>s;
			cnt[s]++;
		}

		map<string, int>::iterator it;
		int answer = 1;
		for(it=cnt.begin(); it!=cnt.end(); it++)
			answer *= (it->second+1);

		cout<<answer-1<<endl;
	}
}

File: JUNGOL/Beginner/#3699 변장/변장.py
def main():
    for i in range(int(input())):
        n = int(input())

        cnt = {}
        for j in range(n):
            t,s = input().split()
            if s in cnt: cnt[s] += 1
            else: cnt[s] = 1

        answer = 1
        for a,b in cnt.items():
            answer *= (b+1)

        print(answer-1)

main()
